analysis/cpp/src/make-histograms-loop.cpp

```diff
@@ -60,33 +60,58 @@ int main(int argc, char* argv[]) {
+    int* mcgst = new int[mc_particle_array_size]; // MC generator status (1 == generated particle, 0 = decay ??)
+    float* mc_px = new float[mc_particle_array_size]; // x momentum
+    float* mc_py = new float[mc_particle_array_size]; // y momentum
+    float* mc_vtx_xpos = new float[mc_particle_array_size]; // x coordinate of MC particle vertex
+    float* mc_vtx_ypos = new float[mc_particle_array_size]; // y coordinate of MC particle vertex
+    float* mc_vtx_zpos = new float[mc_particle_array_size]; // z coordinate of MC particle vertex
+    int n_mc_particles_in_event = 0; // the number of MC particles in the event
 
     // attach the local variables to those in the TTree
+     tree->SetBranchAddress("nmcp", &n_mc_particles_in_event);
     tree->SetBranchAddress("vtxxx", &pv_xpos);
-    tree->SetBranchAddress("mcvtx", &pv_xpos);
-    // tree->SetBranchAddress("vtyyy", &pv_ypos);
-    // tree->SetBranchAddress("vtzzz", &pv_zpos);
-    // tree->SetBranchAddress("mcpdg", mcpdg);
-    // tree->SetBranchAddress("nmcp", &n_mc_particles_in_event);
-    // tree->SetBranchAddress("mcmox", mc_px);
-    // tree->SetBranchAddress("mcmoy", mc_py);
+    tree->SetBranchAddress("mcvtx", mc_vtx_xpos);
+//tree->SetBranchAddress("nmcp", &n_mc_particles_in_event); // we need to load the "nmcp" varialbe so that we know how many MC particles there are in an event
+     //tree->SetBranchAddress("vtyyy", &pv_ypos);
+     //tree->SetBranchAddress("vtzzz", &pv_zpos);
+     tree->SetBranchAddress("mcpdg", mcpdg);
+     tree->SetBranchAddress("mcgst", mcgst);
+     //tree->SetBranchAddress("mcmox", mc_px);
+     //tree->SetBranchAddress("mcmoy", mc_py);
 
     // instantiate the histograms that we want to make
     std::unique_ptr<TH1F> h_vtxxx = std::make_unique<TH1F>("h_vtxxx", "Reco PV x-position;PV x-position [#mum];Entries/bin", 50, -5, 5);
     std::unique_ptr<TCanvas> c_vtxxx = std::make_unique<TCanvas>("c_vtxxx");
-    std::unique_ptr<TH1F> h_mcvtx = std::make_unique<TH1F>("h_mcvtx", "MC PDG Id.;PDG Id.;Entries/bin", 50, -5, 5);
-    std::unique_ptr<TCanvas> c_mcvtx = std::make_unique<TCanvas>("c_mcvtx");
+    std::unique_ptr<TH1F> h_mcvtx_muon = std::make_unique<TH1F>("h_mcvtx_muon", "Muon Truth-level Vertex x-position;Vertex x-position [#mum];Entries/bin", 50, -5, 5);
+    std::unique_ptr<TCanvas> c_mcvtx_muon = std::make_unique<TCanvas>("c_mcvtx_muon");
+
+    std::unique_ptr<TH1F> h_mu_mcvtx_diff = std::make_unique<TH1F>("h_mu_mcvtx_diff", "Difference between Truth-level Muon and Anti-Muon Production Vertices;#Delta Vertex x-position [#mum];Entries/bin", 20, -0.2, 0.2);
+    std::unique_ptr<TCanvas> c_mu_mcvtx_diff = std::make_unique<TCanvas>("c_mu_mcvtx_diff");
+
+    std::unique_ptr<TH1F> h_mc_muon_count = std::make_unique<TH1F>("h_mc_muon_count", "Number of MC (anti-)muons per event;Number of (anti-)muons;Entries/bin", 10, -5, 5);
+    std::unique_ptr<TCanvas> c_mc_muon_count = std::make_unique<TCanvas>("c_mc_muon_count");
```

Comment:
Here I've created a few other histograms:

- `h_mcvtx_muon`: This is the MC vertex position of the muons (not anti-muons) who have generator status == 1
- `h_mu_mcvtx-diff`: This histogram will be filled with the _difference_ between the MC vertex position of the muon and anti-muon in the event -- this will act as a way for us to confirm that the production vertices of the muon and anti-muon in the event are the same or not (they _should_ be the same, since in our muon particle gun we generate both of them at `(0,0,0)`)
- `h_mc_muon_count`: This histogram will be filled with the number of muons and anti muons with generator status `==1` in the event, with muons (`mcpdg == 13`) filling the values `+1.5` in the histogram and antimuons (`mcpdg == -13`) filling the values `-1.5`. This histogram will allow us to see that there are exactly 1000 of each species (since we generate a sample with 1000 events, and each event has both a muon and antimuon).